14 receive multiple requests

jaxdid opened the pull request:
closes #14

jaxdid committed:
WIP User Story #14

jaxdid committed:
Completed User Story 14

- Added validation to prevent users from making requests of their own
  spaces. Disabled request form for space owners.
- Updated request controller logic to automatically deny requests
  conflicting with confirmed booking dates.
- Disabled Requests button for logged out users.
- Disabled confirm option for denied and confirmed requests.
- Updated `booking_spec.rb` to reflect inability of users to request
  their own spaces.

russellvaughan commented on app/app.rb:
```diff
@@ -69,7 +69,14 @@ def validate_dates(space)
+      validate_space(space)
       validate_dates(space)
       booking_from = space.bookings.map(&:from_date)
       booking_to = space.bookings.map(&:end_date)
```
just a little thing but is it maybe worth renaming one of this to something a  little bit different?  Validate_space(space) and validate(space) are a little bit hard to follow?

jaxdid committed:
Refactored User Story 14

- Renamed `#validate_dates` and `#check_bookings` helper methods for
  clarity.
- Refactored `#validate` to simplify date validation method calls.
- Extracted all helper methods from `app/app.rb` to separate
  `helpers.rb` file.
- Updated Request model to make date fields required.
- Updated `views/spaces.erb` and `views/view_space.erb` to make date
  inputs required for form submission.

RPiper93 commented:
:shipit:
